filament/src/main/java/net/fabricmc/filament/task/MapJarTask.java

```diff
@@ -0,0 +1,131 @@
+package net.fabricmc.filament.task;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.FileSystemLocation;
+import org.gradle.api.file.FileVisitDetails;
+import org.gradle.api.file.FileVisitor;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Classpath;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.workers.WorkAction;
+import org.gradle.workers.WorkParameters;
+import org.gradle.workers.WorkQueue;
+import org.gradle.workers.WorkerExecutor;
+
+import net.fabricmc.tinyremapper.OutputConsumerPath;
+import net.fabricmc.tinyremapper.TinyRemapper;
+import net.fabricmc.tinyremapper.TinyUtils;
+
+public abstract class MapJarTask extends DefaultTask {
+	@InputFile public abstract RegularFileProperty getInput();
+	@InputFile public abstract RegularFileProperty getMappings();
+	@OutputFile public abstract RegularFileProperty getOutput();
+	@Classpath public abstract ConfigurableFileCollection getClasspath();
```

Comment:
I didnt know @Classpath is a thing 👍

Reply:
Heh, see the table here: <https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks>

There are some other interesting ones like `@CompileClasspath` and `@Console` too